subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Pmd.java

```diff
@@ -289,6 +290,25 @@ public void setIgnoreFailures(boolean ignoreFailures) {
         this.ignoreFailures = ignoreFailures;
     }
 
+    /**
+     * The maximum number of failures to allow before stopping the build.
+     *
+     * If <pre>ignoreFailures</pre> is set, this is ignored and no limit is enforced.
+     */
+    @Input
+    public int getMaxFailures() {
+        return maxFailures;
+    }
+
+    /**
+     * The maximum number of failures to allow before stopping the build.
+     *
+     * If <pre>ignoreFailures</pre> is set, this is ignored and no limit is enforced.
+     */
+    public void setMaxFailures(int maxFailures) {
```

Comment:
This should validate the value. Things like -1 make little sense in this context. Maybe mentioned the default value as well

Reply:
Good suggestion. Added.